utils/src/CMakeLists.txt

```diff
@@ -24,4 +24,5 @@ if (CONFIG_SIDEWALK_CLI)
     zephyr_include_directories(${ZEPHYR_BASE}/../sidewalk/samples/template/include)
 endif() # CONFIG_SIDEWALK_CLI
 
-zephyr_library_sources_ifdef(CONFIG_SIDEWALK_DFU nordic_dfu.c)
+zephyr_library_sources_ifdef(CONFIG_SIDEWALK_DFU_SERVICE_BLE nordic_dfu.c)
+zephyr_library_sources(settings_utils.c)
```

Comment:
minor: consider adding common dfu config for both libs (settings utils is not necessary for non-bootloader applications like sid_dut)

Reply:
Added clause to compile settings_utils only of SIEWLK_DFU is enabled